tla-io/src/main/scala/at/forsyte/apalache/io/OutputManager.scala

```diff
@@ -48,6 +48,21 @@ object OutputManager extends LazyLogging {
         ProfilingFlag -> false
     )
 
+  // For bug report templates as well as the next iteration of error messages, we will need to reference
+  // lines in the original input. This variable stores them.
+  private var sourceLinesOpt: Option[IndexedSeq[String]] = None
+
+  // Should be called only if input is a .tla file
+  def initSourceLines(file: File): Unit =
+    if (sourceLinesOpt.isEmpty) {
+      val src = Source.fromFile(file)
+      try sourceLinesOpt = Some(src.getLines().toIndexedSeq)
+      finally src.close()
+    }
```

Comment:
Why should it only be called on TLA files? Why not also store the lines from other kinds of input?

Reply:
Because JSON file lines don't correspond to anything sensible (in fact, the `location` fields in the JSON reference the original spec's line numbers) and tnt isn't implemented yet.

Reply:
The JSON file lines would correspond to the JSON source data, at least. Couldn't we get an error that related to some data coming from a bit of bad json?

> the location fields in the JSON reference the original spec's line numbers

This makes me wonder whether we are asking for trouble by introducing a new way to count line numbers (outside of the one we are currently using for source mapping.....).

Would the correct logic here not then be to parse the spec from the JSON file that was input and save *that* in memory here?